src/libraries/System.Formats.Tar/tests/TarWriter/TarWriter.WriteEntry.File.Tests.Windows.cs

```diff
@@ -39,41 +29,14 @@ public partial class TarWriter_WriteEntry_File_Tests : TarTestsBase
 
                 if (entry is PaxTarEntry pax)
                 {
-                    Assert.True(pax.ExtendedAttributes.Count >= 4);
-                    Assert.Contains("path", pax.ExtendedAttributes);
-                    Assert.Contains("mtime", pax.ExtendedAttributes);
-                    Assert.Contains("atime", pax.ExtendedAttributes);
-                    Assert.Contains("ctime", pax.ExtendedAttributes);
-
-                    Assert.True(double.TryParse(pax.ExtendedAttributes["mtime"], NumberStyles.Any, CultureInfo.InvariantCulture, out double doubleMTime));
-                    DateTimeOffset actualMTime = ConvertDoubleToDateTimeOffset(doubleMTime);
-                    VerifyTimestamp(info.LastAccessTimeUtc, actualMTime);
-
-                    Assert.True(double.TryParse(pax.ExtendedAttributes["atime"], NumberStyles.Any, CultureInfo.InvariantCulture, out double doubleATime));
-                    DateTimeOffset actualATime = ConvertDoubleToDateTimeOffset(doubleATime);
-                    VerifyTimestamp(info.LastAccessTimeUtc, actualATime);
-
-                    Assert.True(double.TryParse(pax.ExtendedAttributes["ctime"], NumberStyles.Any, CultureInfo.InvariantCulture, out double doubleCTime));
-                    DateTimeOffset actualCTime = ConvertDoubleToDateTimeOffset(doubleCTime);
-                    VerifyTimestamp(info.LastAccessTimeUtc, actualCTime);
+                    VerifyPaxTimestamps(pax);
                 }
 
                 if (entry is GnuTarEntry gnu)
                 {
-                    VerifyTimestamp(info.LastAccessTimeUtc, gnu.AccessTime);
-                    VerifyTimestamp(info.CreationTimeUtc, gnu.ChangeTime);
+                    VerifyGnuTimestamps(gnu);
                 }
             }
         }
-
-        private void VerifyTimestamp(DateTime expected, DateTimeOffset actual)
-        {
-            // TODO: Find out best way to compare DateTime vs DateTimeOffset,
-            // because DateTime seems to truncate the miliseconds https://github.com/dotnet/runtime/issues/68230
-            Assert.Equal(expected.Date, actual.Date);
-            Assert.Equal(expected.Hour, actual.Hour);
-            Assert.Equal(expected.Minute, actual.Minute);
-            Assert.Equal(expected.Second, actual.Second);
-        }
```

Comment:
This was causing issues, particularly when comparing AccessTime that comes from an actual file on disk, which gets modified when it is opened. So in this particular test, I simplified it by making sure the value was at least larger than UnixEpoch in the tests.

I added some other tests in this PR that are verifying the values of the timestamps, but I plan to add more tests for comparing timestamps coming from files specifically, when I address these two issues: https://github.com/dotnet/runtime/issues/70060 and https://github.com/dotnet/runtime/issues/69474

Reply:
I'm bringing this method back but I'm putting it in the base class. It is helpful for when I am converting the DateTimeOffset to double and then back to DateTimeOffset. There seems to be some precision loss in the milliseconds and smaller units.
I added an offset comparison, to make sure the timestamp is getting converted to UTC.

Reply:
Added in the latest commits.

Reply:
While the UTC problem was fixed, the comparisons in the test aren't working. Investigating.

Reply:
The last runtime-extra-platforms run is not hitting the error anymore. The new comparisons seem to work.